### PR TITLE
Refactor to use upstream MHLO targets 

### PR DESCRIPTION
**Pull request by marbre**
* Drops the overlay external_tablegen_library() calls to generate MHLO
  tablegen targets and cleans up the custom hlo CMake config
* Refactors the custom tensorflow_mlir_hlo target to depend on the
  upstream MHLO targets
* Drops the external_tablegen_library CMake module

Depends on https://github.com/tensorflow/tensorflow/pull/44589.

### CODE REVIEW

**Comment by marbre** [comment]
I leave this as a draft PR until we can bump tensorflow, but maybe you @GMNGeoffrey already want to take a look?

**Review by GMNGeoffrey — APPROVED** [review]

**Comment by GMNGeoffrey** [comment]
@benvanik not sure if you want to weigh in on the include dir stuff. I know you want to get rid of IREE_COMMON_INCLUDE_DIRS and it's not clear whether this makes things better or worse in that direction :grin: (https://github.com/google/iree/issues/2195)

**Comment by GMNGeoffrey** [comment]
@marbre please ping this (and maybe re-request my review?) when your TF patch wends its way to `main` so I can merge this

File: CMakeLists.txt [hunk 1]
```diff
@@ -323,17 +327,27 @@ function(add_iree_mlir_src_dep llvm_monorepo_path)
 
   # Reset CMAKE_BUILD_TYPE to its previous setting.
   set(CMAKE_BUILD_TYPE "${_CMAKE_BUILD_TYPE}" CACHE STRING "Build type (default ${DEFAULT_CMAKE_BUILD_TYPE})" FORCE)
+
+  # Set include directories
+  include_directories(${PROJECT_SOURCE_DIR}/third_party/llvm-project/llvm/include)
+  include_directories(${PROJECT_BINARY_DIR}/third_party/llvm-project/llvm/include)
+  include_directories(${PROJECT_SOURCE_DIR}/third_party/llvm-project/mlir/include)
+  include_directories(${PROJECT_BINARY_DIR}/third_party/llvm-project/llvm/tools/mlir/include)
```
Comment (benvanik) [inline CMakeLists.txt:335]: yeah no global includes like this - these should not be needed anyway
adding includes like this adds them to every target for every platform for all of the project - they should be added only where needed

**Review by benvanik — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (marbre) [reply CMakeLists.txt:335]: Well we need to make MLIR-HLO aware of where it finds MLIR. This normally happens via

```CMake
find_package(MLIR REQUIRED CONFIG)
...
include_directories(${LLVM_INCLUDE_DIRS})
include_directories(${MLIR_INCLUDE_DIRS})
```

We cannot use this, as we build a bundled MLIR... or at least I was not able to get it running with a bundled MLIR + passing `MLIR_DIR`. Using an installed MLIR, it's done the way described above. However, we need to pass those includes to some submodules, especially MLIR-HLO. Before this change, the includes were set indirectly via `iree_copts`. Thus, I don't think this change makes it worse, also it is not a big iprovement. Any suggestions who we could do better?

Comment (marbre) [reply CMakeLists.txt]: `MLIR_INCLUDE_DIRS` and `LLVM_INCLUDE_DIRS` are now used to propagate the include directories to submodules that normally rely on `find_package(MLIR REQUIRED CONFIG)`. Furthermore, the changes to `IREE_COMMON_INCLUDE_DIRS` in `iree_copts.cmake` were reverted and the open TODO can be addressed separately.

Comment (benvanik) [reply CMakeLists.txt]: awesome!

**Review by benvanik — APPROVED** [review]

**Comment by marbre** [comment]
> @marbre please ping this (and maybe re-request my review?) when your TF patch wends its way to `main` so I can merge this

The TF patch is available after the latest submodule bump and I rebased the PR onto the current main. Thus it should be ready to merge.

**Pull request merged**